iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveTableUtil.java

```diff
@@ -0,0 +1,142 @@
+import org.apache.iceberg.mapping.NameMappingParser;
+import org.apache.iceberg.relocated.com.google.common.util.concurrent.ThreadFactoryBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class HiveTableUtil {
+
+  private static final Logger LOG = LoggerFactory.getLogger(HiveTableUtil.class);
+
+  private static class DataFileFilter implements PathFilter {
+    private final FileSystem fileSystem;
+
+    private DataFileFilter(FileSystem fileSystem) {
+      this.fileSystem = fileSystem;
+    }
+
+    @Override
+    public boolean accept(Path path) {
+      String name = path.getName();
+      try {
+        return !name.startsWith("_") && !name.startsWith(".") && fileSystem.isFile(path);
+      } catch (IOException e) {
+        return false;
+      }
+    }
+  }
+
+  private HiveTableUtil() {
+  }
+
+  /**
+   * Import files from given partitions to an Iceberg table.
+   * @param sourceLocation location of the HMS table
+   * @param format inputformat class name of the HMS table
+   * @param partitionSpecProxy  list of HMS table partitions wrapped in partitionSpecProxy
+   * @param icebergTable destination iceberg table
+   * @param conf a Hadoop configuration
+   */
+  public static void importFiles(String sourceLocation, String format, PartitionSpecProxy partitionSpecProxy,
+      Table icebergTable, Configuration conf) {
+    AppendFiles append = icebergTable.newAppend();
+    try {
+      URI uri = new URI(sourceLocation);
+      FileSystem fileSystem = FileSystem.get(uri, conf);
+      PartitionSpec spec = icebergTable.spec();
+      MetricsConfig metricsConfig = MetricsConfig.fromProperties(icebergTable.properties());
+      String nameMappingString = icebergTable.properties().get(TableProperties.DEFAULT_NAME_MAPPING);
+      NameMapping nameMapping = nameMappingString != null ? NameMappingParser.fromJson(nameMappingString) : null;
+      if (partitionSpecProxy.size() == 0) {
+        List<DataFile> dataFiles = getDataFiles(fileSystem, new Path(uri), Collections.emptyMap(), format, spec,
+            metricsConfig, nameMapping, conf);
+        dataFiles.forEach(append::appendFile);
+      } else {
+        PartitionSpecProxy.PartitionIterator iterator = partitionSpecProxy.getPartitionIterator();
+        List<Callable<List<DataFile>>> tasks = new ArrayList<>();
+        while (iterator.hasNext()) {
+          Partition partition = iterator.next();
+          Callable<List<DataFile>> task = () -> {
+            Path partitionPath = new Path(partition.getSd().getLocation());
+            Map<String, String> partitionSpec = Warehouse.makeSpecFromName(partitionPath.toString());
+            return getDataFiles(fileSystem, partitionPath, partitionSpec, format.toLowerCase(), spec, metricsConfig,
+                nameMapping, conf);
+          };
+          tasks.add(task);
+        }
+        int numThreads = HiveConf.getIntVar(conf, HiveConf.ConfVars.HIVE_SERVER2_ICEBERG_METADATA_GENERATOR_THREADS);
```

Comment:
Maybe not worth it, just wondering... Do we want to have a separate codepath where the threadnum is 1?